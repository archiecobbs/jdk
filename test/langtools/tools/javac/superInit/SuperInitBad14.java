/*
 * @test /nodynamiccopyright/
 * @bug 8194743
 * @summary Disallow calls to this() and super() within try blocks
 *
 * @compile/fail/ref=SuperInitBad14.out -XDrawDiagnostics SuperInitBad14.java
 */
import java.io.*;
class SuperInitBad14 {

    SuperInitBad14() {
        try {
            this(4);
        } catch (IllegalArgumentException e) {
            throw new RuntimeException();
        }
    }

    SuperInitBad14(int x) {
        try {
            super();
        } catch (Error e) {
            throw new IllegalArgumentException();
        }
    }

    SuperInitBad14(char x) {
        try {
            this((int)x);
        } finally {
            throw new IllegalArgumentException();
        }
    }

    SuperInitBad14(short x) throws IOException {
        try (StringWriter w = new StringWriter()) {
            this((int)x);
        }
    }
}
