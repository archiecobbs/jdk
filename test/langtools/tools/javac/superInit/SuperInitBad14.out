SuperInitBad14.java:13:17: compiler.err.call.may.not.appear.within.try: this
SuperInitBad14.java:21:18: compiler.err.call.may.not.appear.within.try: super
SuperInitBad14.java:29:17: compiler.err.call.may.not.appear.within.try: this
SuperInitBad14.java:37:17: compiler.err.call.may.not.appear.within.try: this
4 errors
